crypto/async/arch/async_posix.c

```diff
@@ -28,6 +50,43 @@ int ASYNC_is_capable(void)
     return getcontext(&ctx) == 0;
 }
 
+int ASYNC_set_mem_functions(ASYNC_stack_alloc_fn alloc_fn,
+                            ASYNC_stack_free_fn free_fn)
+{
```

Comment:
Needs a call to `OPENSSL_init_crypto(OPENSSL_INIT_ASYNC, NULL)` here to make sure the lock has actually been initialised.